src/Context/GtmContext.php

```diff
@@ -77,16 +78,70 @@ public function getDataLayerSettingShouldMatch($key, $regex) {
    */
   protected function getDataLayerValue($key) {
     $json_arr = $this->getDataLayerJson();
-
-    // Loop through the array and return the data layer value
+    // Loop through the array and return the data layer value.
     foreach ($json_arr as $json_item) {
-      if (isset($json_item[$key])) {
-        return $json_item[$key];
+      // Check if the key contains dot.
+      if (strpos($key, '.', 0) !== false) {
+        // Get value using dot notation.
+        $value = $this->getDotValue($json_item, $key);
+        if (!is_null($value)) {
+          return $value;
+        }
+      } elseif (isset($json_item[$key])) {
+          return $json_item[$key];
       }
     }
     throw new \Exception($key . ' not found.');
   }
 
+  /**
+   * Convert arrays to dot notation
+   * @param $value
+   * @param $key
+   * @return mixed|void
+   */
+  private function getDotValue($value, $key) {
+    $dot = dot($value);
```

Comment:
Please delete this line

Reply:
Can't delete this because in next line we are using this dot function 
`$values = $dot->get($keyBeforeArray);`

Reply:
Ok, then I will suggest another change below on line 122